experimental/Origami/src/Origami.jl

```diff
@@ -0,0 +1,299 @@
+    for i in 1:different_cycle_count
+        cycle_type_entry = cycle_struc[i]
+        cycle_type_length = cycle_struc[i][1]
+        if(cycle_type_entry[1] == 1)
+            # ignore cycles of length 1
+            continue
+        end
+        for j in 1:cycle_type_entry[2]
+            push!(stratum, ZZ(cycle_type_length - 1))
+        end
+    end
+    return stratum
+end
+
+function genus(o::Origami)
+    return ZZ((sum(stratum(o)) + 2) * 0.5)
+end
+
+function veech_group(O::Origami)
+    # TODO use hashtables or not? Implement modular subgroup?
+    GAP.Globals.ComputeVeechGroupWithHashTables(GapObj(O))
+end
+
+function index_monodromy_group(o::Origami)
+    GAP.Globals.IndexOfMonodromyGroup(GapObj(o))
+end
+
+function sum_of_lyapunov_exponents(o::Origami)
+    gap_obj = GAP.Globals.SumOfLyapunovExponents(GapObj(o))
+    return GAP.gap_to_julia(gap_obj)
+end
+
+function translations(o::Origami)
+    h = horizontal_perm(o)
+    v = vertical_perm(o)
+    G = normalform_conjugators(o)
+
+    O = (i -> origami(i^-1 * h * i, i^-1 * v * i)).(G)
+    list = [cperm([degree(o)])]
+
+    l = length(O)
+    for i in 1:l
+        positions = findall(item -> item == O[i], O)
+        if length(positions) != 1
+            for j in positions
+                push!(list, G[i] * G[j]^-1)
+            end
+        end
+    end
+
+    return collect(Set(list))
+end
+
+function is_hyperelliptic(o::Origami)
+    # check whether -1 is in the veech group
+	if !veech_group_is_even(o) 
+        return false
+    end
+
+    n = 2
+    n_inv = 1 / 2
+	x = horizontal_perm(o)
+	y = vertical_perm(o)
+	g = genus(o)
+
+    L = point_reflections(o)
+	L = filter(i -> order(i) == 2, L)
+
+	if L == []
+        return false
+    end
+
+    # helper to find all elements in x that are not in y
+    function list_diff(x, y)
+        set_1 = Set(x)
+        set_2 = Set(y)
+        diff = setdiff(set_1, set_2)
+        return collect(diff)
+    end
+
+    degree_list = collect(1:degree(o))
+    for sigma in L
+        # fixpoints
+        b = 0
+        b = b + length(list_diff(degree_list, moved_points(sigma)))
+        b = b + length(list_diff(degree_list, moved_points(sigma*x)))
+        b = b + length(list_diff(degree_list, moved_points(sigma*y)))
+
+        for i in degree_list
+            if i^(sigma * x^-1 * y^-1) == i^(y*x*(x*y)^-1)
+                b = b + 1
+            end
+        end
+
+        # TODO can n_inv yields floating point errors?
+        if n_inv * (g - 1 - b * 0.5) + 1 == 0
+            return true
+        end
+    end
+
+    return false
+end
+
+function cylinder_structure(o::Origami)
+    # TODO returns Vector{Any}, maybe cast to Integer?
+    gap_obj = GAP.Globals.CylinderStructure(GapObj(o))
+    return GAP.gap_to_julia(gap_obj)
+end
+
+function veech_group_and_orbit(o::Origami)
+    return GAP.Globals.VeechGroupAndOrbit(GapObj(o))
+end
+
+function veech_group_is_even(o::Origami)
+    return GAP.Globals.VeechGroupIsEven(GapObj(o))
```

Comment:
Does this return a boolean? Then I suggest adding a type annotation to make this clear, and to make the code type stable:
```suggestion
    return GAP.Globals.VeechGroupIsEven(GapObj(o))::Bool
```
Also, I would then rename this function to `is_veech_group_even`, to match our general naming conventions.